rules/hardcoded_credentials.go

```diff
@@ -79,12 +81,24 @@ func (r *credentials) matchAssign(assign *ast.AssignStmt, ctx *gosec.Context) (*
 					}
 				}
 			}
+
+			// Match the value assigned
+			for _, e := range assign.Rhs {
+				if val, err := gosec.GetString(e); err == nil {
+					if r.patternValue.MatchString(val) {
+						if r.ignoreEntropy || (!r.ignoreEntropy && r.isHighEntropyString(val)) {
```

Comment:
`!r.ignoreEntropy` will always be `true` here, since it'll only get evaluated if `r.ignoreEntropy` is false. The expression can be simplified to `r.ignoreEntropy || r.isHighEntropyString(val)`

Reply:
Nice catch! I resolved these short-circuit evaluation suggestions.